lib/vrl/vrl/src/runtime.rs

```diff
@@ -98,20 +98,24 @@ impl Runtime {
             }
         };
 
-        let mut context = Context::new(target, &mut self.state, timezone);
+        let mut ctx = Context::new(target, &mut self.state, timezone);
 
-        let mut values = program
-            .iter()
-            .map(|expr| {
-                expr.resolve(&mut context).map_err(|err| match err {
-                    #[cfg(feature = "expr-abort")]
-                    ExpressionError::Abort { .. } => Terminate::Abort(err),
-                    err @ ExpressionError::Error { .. } => Terminate::Error(err),
-                })
-            })
-            .collect::<Result<Vec<_>, _>>()?;
-
-        Ok(values.pop().unwrap_or(Value::Null))
+        let err = |err| match err {
+            #[cfg(feature = "expr-abort")]
+            ExpressionError::Abort { .. } => Terminate::Abort(err),
+            err @ ExpressionError::Error { .. } => Terminate::Error(err),
+        };
+
+        match program.split_last() {
+            None => Ok(Value::Null),
```

Comment:
This differs slightly from a block now. A block assumes it can't be empty, but a program can be empty? If this is switched to a block, make sure it behaves similarly